src/pages/home/report/ReportActionCompose.js

```diff
@@ -114,7 +126,9 @@ class ReportActionCompose extends React.Component {
      */
     focus() {
         if (this.textInput) {
-            this.textInput.focus();
+            InteractionManager.runAfterInteractions(() => {
+                this.textInput.focus();
+            });
```

Comment:
I'm confused about what issue this is addressing. Can we add some comments to explain what interactions we are waiting to finish?

Reply:
Animations do not run smoothly, as a side effect of the focus. So this case should suffice when the modal closes. It's a precautionary check that's all. Not particularly needed for these changes.

Reply:
Cool whatever it is, please add an explanation so we have some insights when looking at the code. Out of curiositym is there a particular modal closing that causes jank? Maybe we can add a test to verify it does not look janky?

cc @Julesssss We should also maybe update these test steps so that it is testable by QA?

Reply:
So, the Most effect of focus interfering with animation can be seen on a new chat or new group screen where we set the focus on Search input. Here, we may need this on the web when we close modals and then set focus back to Input but it's not apparent. Due to the similar effect happening on New Chat modals, I thought it's better to apply `Interactionmanager` on all the places where we are setting focus.

Reply:
> So, the Most effect of focus interfering with animation can be seen on a new chat or new group screen where we set the focus on Search input.

But these are due to the `autoFocus` the logic would be applied for manual/programmatic focuses only
If you want to make auto focus wait for interactions as well you'll need to remove the `autoFocus` prop and call `this.focus()` when the component mount

> I thought it's better to apply `Interactionmanager` on all the places where we are setting focus.

This applies it only here on compose though right? Opening a new chat or search use the `TextInputFocusable`
Instead of adding this logic to all the places that set focus couldn't it perhaps be added to `TextInputFocusable` instead.
`TextInputFocusable` can wrap `inputRef.focus` with this behavior (before exposing it to parents)

Since focusing is done inside a callback, is it guaranteed you will always be inside a chat when this callback is invoked. Could it be possible for the user to tap/swipe open a chat and go back to LHN or somewhere else and then this callback runs and focuses the field
- `if (this.textInput)` - `textInput` - could be defined before setting this interaction, but could have changed inside the actual callback
- you will want to abort this interaction in case the component is unmounted but the callback is still scheduled to run. Similar to how you would clean up timers when a component unmounts

```js
focus() {
  this.scheduledIntercation = Interactionmanager.runAfterInteractions(() => {...})
}

componentWillUnmount() {
  if(this.scheduledIntercation) this.scheduledIntercation.cancel()
}
```

I think this works great, just that it should have been reusable/configurable for the other input fields as well